### DISCUSSION TITLE
Make import action tenant aware

### DISCUSSION THREAD

**Discussion by adesege** [filamentphp/filament#11224] [Ideas]
Hi,

I have a multi-tenancy setup and I'm trying to access `Filament::getTenant()` inside an importer class. The method returns null.

Upon further investigation, the route `filament.imports.failed-rows.download` is only registered at the top level.

It will be nice to make import action tenant-aware so that tenancy methods can be used within an importer class

**Comment by ronssij** [comment]
@adesege found a workound on this inside the Importer class.

```php
/**
 * @method User getRecord()
 */
class PatientImporter extends Importer
{
    protected static ?string $model = User::class;

    public $tenant = null;

    public function __construct(
        protected Import $import,
        protected array $columnMap,
        protected array $options,
    ) {
        $this->tenant = filament()->getTenant();
    }

   protected function afterCreate(): void
   {
        $this->getRecord()->syncToCurrentTenant($this->tenant);
   }
}
